generators/php/codegen/src/context/AbstractPhpGeneratorContext.ts

```diff
@@ -218,6 +245,115 @@ export abstract class AbstractPhpGeneratorContext<
+            case "list": {
+                return typeReference.container.list._visit({
+                    container: (container) => {
+                        if (this.isCollection(TypeReference.container(container))) {
+                            return this.dereferenceCollection(TypeReference.container(container));
+                        } else {
+                            return TypeReference.container(container) as TypeReference;
+                        }
+                    },
+                    named: (named) => {
+                        return TypeReference.named(named) as TypeReference;
+                    },
+                    primitive: (primitive) => {
+                        return TypeReference.primitive(primitive) as TypeReference;
+                    },
+                    unknown: () => {
+                        return TypeReference.unknown() as TypeReference;
+                    },
+                    _other: () => {
+                        return TypeReference.unknown() as TypeReference;
+                    }
+                });
+            }
+            case "set": {
+                return typeReference.container.set._visit({
+                    container: (container) => {
+                        if (this.isCollection(TypeReference.container(container))) {
+                            return this.dereferenceCollection(TypeReference.container(container));
+                        } else {
+                            return TypeReference.container(container) as TypeReference;
+                        }
+                    },
+                    named: (named) => {
+                        return TypeReference.named(named) as TypeReference;
+                    },
+                    primitive: (primitive) => {
+                        return TypeReference.primitive(primitive) as TypeReference;
+                    },
+                    unknown: () => {
+                        return TypeReference.unknown() as TypeReference;
+                    },
+                    _other: () => {
+                        return TypeReference.unknown() as TypeReference;
+                    }
+                });
+            }
+            default: {
+                return typeReference;
+            }
+        }
+    }
+
+    public isCollection(typeReference: TypeReference): boolean {
+        return (
+            typeReference.type === "container" &&
+            (typeReference.container.type === "list" || typeReference.container.type === "set")
+        );
+    }
+
+    public isJsonEncodable(typeReference: TypeReference): boolean {
+        return (
+            typeReference.type === "unknown" ||
+            (typeReference.type === "container" && typeReference.container.type === "map")
+        );
+    }
+
+    public hasToJsonMethod(typeReference: TypeReference): boolean {
+        return typeReference.type === "named" && !this.isPrimitive(typeReference) && !this.isEnum(typeReference);
+    }
```

Comment:
Same note with the `isOptional` helper for all these - we want to make sure we follow aliases too.

Reply:
doe